### PR TITLE
feat: Add ability to close registration completely

### LINKED ISSUE #169: Add ability to close registration entirely
_Copied from https://github.com/codeRIT/brickhack.io/issues/241_

>A few people are actually still applying, we should completely close registration to eliminate confusion until we re-open for 2016.

There should be a new configuration (true/false) field to completely prevent new people from registering (maybe `disable_account_registration`)

This will likely involve hooking in with Devise and overriding the default Devise controller, something like what's described [here](https://groups.google.com/forum/#!topic/plataformatec-devise/pk3CptTL8kw) but instead of `def new` always doing a redirect, conditionally do the flash + redirect and otherwise call `super`

### PR DESCRIPTION
**Pull request by dquach217**
We would check if the config is true or not, if they try to register it would redirect to the root path. 

Fixes #169

### CODE REVIEW

File: app/controllers/users/registrations_controller.rb [hunk 1]
```diff
@@ -3,9 +3,14 @@ class Users::RegistrationsController < Devise::RegistrationsController
   # before_action :configure_account_update_params, only: [:update]
 
   # GET /resource/sign_up
-  # def new
-  #   super
-  # end
+  def new
+    if HackathonConfig['disable_account_registration']
+      flash[:alert] = "Registration has been closed"
```
Comment (sman591) [inline app/controllers/users/registrations_controller.rb]:
```suggestion
      flash[:alert] = "Registration has closed"
```

**Comment by dquach217** [comment]
@sman591 Updated so the check happens before new and create. Please review when you get the chance to.

File: app/controllers/manage/questionnaires_controller.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 class Manage::QuestionnairesController < Manage::ApplicationController
   include QuestionnairesControllable
 
+  before_action :check_for_registration, only: [:new, :create]
```
Comment (sman591) [inline app/controllers/manage/questionnaires_controller.rb]:
```suggestion
  before_action :ensure_registration_is_open, only: [:new, :create]
```

File: app/controllers/manage/questionnaires_controller.rb [hunk 3]
```diff
@@ -169,4 +170,10 @@ def convert_boarded_bus_param(values, questionnaire = nil)
   def set_questionnaire
     @questionnaire = ::Questionnaire.find(params[:id])
   end
+
+  def check_for_registration
+    if HackathonConfig['disable_account_registration']
+      flash[:alert] = "Registration has closed"
+      redirect_to root_path
+    end
```
Comment (sman591) [inline app/controllers/manage/questionnaires_controller.rb]: Another `end` is needed here to close the `def` block

File: config/locales/en.yml [hunk 4]
```diff
@@ -30,7 +30,6 @@
 # available at http://guides.rubyonrails.org/i18n.html.
 
 en:
-  devise:
```
Comment (sman591) [inline config/locales/en.yml]: Oh good catch

**Comment by dquach217** [comment]
@sman591 Finished applying changes that was requested.

File: app/controllers/manage/questionnaires_controller.rb [hunk 5]
```diff
@@ -169,4 +170,11 @@ def convert_boarded_bus_param(values, questionnaire = nil)
   def set_questionnaire
     @questionnaire = ::Questionnaire.find(params[:id])
   end
+
+  def check_for_registration
```
Comment (sman591) [inline app/controllers/manage/questionnaires_controller.rb]:
```suggestion
  def ensure_registration_is_open
```

**Comment by dquach217** [comment]
@sman591 one test is failing, but not sure why

**Comment by sman591** [comment]
Looks like it's a failing test from #178 -- I just pushed a fix, if you rebase it should resolve

**Comment by dquach217** [comment]
@sman591 Hmmm I see that the test failed when creating the db. Not sure if we want to resolve this or not.

**Comment by sman591** [comment]
Could you pull the latest master and then rebase? It looks like this is still on an old master

**Comment by dquach217** [comment]
@sman591 just rebased it but issue now lies with codeclimate

**Comment by dquach217** [comment]
Should the resolution be to add tests then?

**Pull request merged**